src/validation/coinbase.cairo

```diff
@@ -31,7 +33,40 @@ pub fn validate_coinbase(
     assert(total_output_amount <= total_fees + block_reward, 'total output > block rwd + fees');
 
     // TODO: validate BIP-34 sig script
-    // TODO: validate BIP-141 witness field
+
+    // validate BIP-141 witness field
+    if block_height >= BIP_141_BLOCK_HEIGHT {
+        let witness_arr = *tx.inputs[0].witness;
+        let witness = witness_arr[0];
+
+        // check witness array element
+        if witness_arr.len() > 1 {
+            return Result::Err("witness array should contain only one element");
+        }
+
+        // check witness byte lenght
+        if witness.len() != 32 {
+            return Result::Err("Wrong witness length");
+        };
+
+        // check witness value
```

Comment:
ByteArray has `PartialEq` you can construct the expected value with `"\0\0\0\0\0\0\0\0\0..."` and then compare as usual

Reply:
No need to additionally check for the length then